File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length-1;
        int val = 0;
        while (left < right){
            int curr = Math.min(heights[left] ,heights[right] );
            val = Math.max(val,curr*(right-left));
            if (heights[left]<heights[right] ){
                left++;
            }
            else{
                right --;
            }
        }

        return val;

    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        maxLen = 0
        freq = {}
        left = 0
        maxFreq = 0
        for right in range(len(s)):
            freq[s[right]] = 1 + freq.get(s[right],0)

            maxFreq = max(maxFreq, freq[s[right]])
            if (right - left + 1) - maxFreq > k:
                # shrink window
                freq[s[left]] -= 1
                left += 1

            # update answer
            maxLen = max(maxLen, right - left + 1)



        return maxLen
